airflow/providers/google/cloud/hooks/cloud_build.py

```diff
@@ -55,95 +52,564 @@ class CloudBuildHook(GoogleBaseHook):
+        """
+        Creates a new build based on the specified build. This method creates a new build
+        using the original build request, which may or may not result in an identical build.
 
-        result = (
-            service.projects()
-            .builds()
-            .get(projectId=project_id, id=build_id)
-            .execute(num_retries=self.num_retries)
+        :param id: Build ID of the original build.
+        :type id: str
+        :param project_id: Optional, Google Cloud Project project_id where the function belongs.
+            If set to None or missing, the default project_id from the GCP connection is used.
+        :type project_id: str
+        :param wait: Optional, wait for operation to finish.
+        :type wait: Optional[bool]
+        :param retry: Optional, a retry object used  to retry requests. If `None` is specified, requests
+            will not be retried.
+        :type retry: Optional[Retry]
+        :param timeout: Optional, the amount of time, in seconds, to wait for the request to complete.
+            Note that if `retry` is specified, the timeout applies to each individual attempt.
+        :type timeout: Optional[float]
+        :param metadata: Optional, additional metadata that is provided to the method.
+        :type metadata: Optional[Sequence[Tuple[str, str]]]
+
+        :rtype: `google.cloud.devtools.cloudbuild_v1.types.Build`
+        """
+        client = self.get_conn()
+
+        self.log.info("Start retrying build: %s.", id)
+
+        operation = client.retry_build(
+            request={'project_id': project_id, 'id': id}, retry=retry, timeout=timeout, metadata=metadata
         )
 
-        return result
-
-    def _wait_for_operation_to_complete(self, operation_name: str) -> None:
-        """
-        Waits for the named operation to complete - checks status of the
-        asynchronous call.
-
-        :param operation_name: The name of the operation.
-        :type operation_name: str
-        :return: The response returned by the operation.
-        :rtype: dict
-        :exception: AirflowException in case error is returned.
-        """
-        service = self.get_conn()
-        while True:
-            operation_response = (
-                service.operations().get(name=operation_name).execute(num_retries=self.num_retries)
-            )
-            if operation_response.get("done"):
-                response = operation_response.get("response")
-                error = operation_response.get("error")
-                # Note, according to documentation always either response or error is
-                # set when "done" == True
-                if error:
-                    raise AirflowException(str(error))
-                return response
-            time.sleep(TIME_TO_SLEEP_IN_SECONDS)
+        try:
+            id = operation.metadata.build.id
+        except Exception:
+            raise AirflowException("Could not retrieve Build ID from Operation.")
```

Comment:
We are doing it at least in two operators. Do you think it would worth to introduce `get_id_from_operation` method in hook to keep this behaviour consistent?